torch/nn/modules/transformer.py

```diff
@@ -174,22 +191,28 @@ def __init__(self, decoder_layer, num_layers, norm=None):
         self.num_layers = num_layers
         self.norm = norm
 
-    def forward(self, tgt, memory, tgt_mask=None, memory_mask=None):
+    def forward(self, tgt, memory, mask=None,
+                memory_mask=None, key_padding_mask=None,
+                memory_key_padding_mask=None):
         r"""Pass the inputs (and mask) through the decoder layer in turn.
 
         Args:
             tgt: the sequence to the decoder (required).
             memory: the sequnce from the last layer of the encoder (required).
-            tgt_mask: the mask for the tgt sequence (optional).
+            mask: the mask for the tgt sequence (optional).
             memory_mask: the mask for the memory sequence (optional).
+            key_padding_mask: the mask for the tgt keys per batch (optional).
```

Comment:
For consistency with Transformer, I think we can use "tgt_key_padding_mask" here, instead of key_padding_mask. Sorry for being annoying.